sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesWalker.cs

```diff
@@ -81,11 +97,31 @@ private void FindExecutableLines(SyntaxNode node)
 
                 case SyntaxKind.ArrayInitializerExpression:
                     executableLineNumbers.Add(node.GetLocation().GetLineNumberToReport());
-                    return;
+                    return true;
 
                 default:
-                    return;
+                    return true;
             }
         }
+
+        private static bool HasExcludedCodeAttribute(SyntaxNode node)

```

Comment:
We usually check with the type (semantic model) rather than string so I would do it here too.

Reply:
We only checked the syntax model for auto-generated code when looking for attributes. I guess it could be a performance issue if we started looking up for the symbol of a lot of things. I'll leave it like that for now.